internal/datasource/config/config_writer.go

```diff
@@ -100,25 +99,44 @@ func (cw *ConfigWriter) Write(ctx context.Context, filesURL,
 	}
 
 	for _, fileData := range filesMetaData.GetFiles() {
-		if !doesFileRequireUpdate(cacheContent, fileData) {
+		if cacheContent != nil && !doesFileRequireUpdate(cacheContent, fileData) {
 			slog.Debug("Skipping file as latest version is already on disk", "file_path", fileData.GetPath())
 			currentFileCache[fileData.GetPath()] = cacheContent[fileData.GetPath()]
 			skippedFiles[fileData.GetPath()] = fileData
 
 			continue
 		}
 		slog.Debug("Updating file, latest version not on disk", "file_path", fileData.GetPath())
-		updateErr := cw.updateFile(ctx, fileData, filesURL, tenantID, instanceID)
-		if updateErr != nil {
+		if updateErr := cw.updateFile(ctx, fileData, filesURL, tenantID, instanceID); updateErr != nil {
 			return nil, updateErr
 		}
+		currentFileCache[fileData.GetPath()] = fileData
+	}
+
+	if cacheContent != nil {
+		cw.removeFiles(currentFileCache, cacheContent)
 	}
 
 	cw.currentFileCache = currentFileCache
 
 	return skippedFiles, err
 }
 
+func (cw *ConfigWriter) removeFiles(currentFileCache, fileCache CacheContent) {
+	for _, file := range fileCache {
+		_, ok := currentFileCache[file.GetPath()]
+		if !ok {
+			slog.Debug("removing file", "file_path", file.GetPath())
+			err := os.Remove(file.GetPath())
+			if err != nil {
+				slog.Error("error removing file: ", "err", err)
```

Comment:
If we fail to delete a file then we should return an error from this function. A config apply should fail if we can't update or delete files.

Reply:
Ok, I wasn't sure if we should be failing the config apply if we can't remove a file. I'll change it back to returning an error from that function.